charliescheer/InAppNotifications
# Initial mockup of in app notifications

This PR is to showcase a early sketch of what in app notifications could be.

In this test project, when loaded, you will see a text view with a bunch of text in it and two buttons in the nav bar at the top "Alert" and "Action"

Both of these buttons are triggering the in app notifications, alert is just a message and action is a message and an alert.
A few things to note:
Only one notification can be triggered at a time
If multiple notifications are triggered they will continue to appear until all notifications have been viewed
Notifications will disappear on their own, notifications w/o a button disappear faster
If view is dismissed the alert persists

This is intended to be very early steps, so there are some keys to this project that are not currently available:
in app notifications do not currently support rotating
the notifications do not currently account for the keyboard being displayed or not
They will disappear regardless of it they are interacted with or not
The buttons have not been styled, currently they are styled to be easily visible in the tests

<img width="300px" src="https://cdn-std.droplr.net/files/acc_593859/hTevuG" />

To test:
1. load test app, and tap on the Load Next View Button
2. Tap on the Alert button and see the in app notification without a button appear, and wait for it to fade
3. tap on the Action button to see the message with an action appear
4. before it disappears tap on the button in the notification, a message will appear in Xcode's console
5. let the action disappear
6. tap on Action, then tap on alert while the original notification is still visible.  When the first one disappears the second one will appear automatically
7. Tap on action, once the notification appears hit the back button to make sure that the notification persists after the current view is dismissed

I have attempted to start building this to be able to be used in more than just this instance, so there are some options sketched in that won't apply to this project, but may be useful later (like adding the queue protocol or allowing the notifications to have a top to bottom/bottom to top option)
